ord_schema/orm/database.py

```diff
@@ -93,26 +94,70 @@ def delete_dataset(dataset_id: str, session: Session) -> None:
     logger.info(f"delete took {time.time() - start}s")
 
 
-def add_rdkit(session: Session) -> None:
-    """Adds RDKit PostgreSQL cartridge data to any null-valued cells in the structure table."""
-    logger.info("Populating RDKit reaction columns")
+def update_rdkit(dataset_id: str, session: Session) -> None:
+    """Updates RDKit PostgreSQL cartridge data."""
+    # select distinct smiles, count(*) from rdkit.mols where smiles similar to '\[[A-Z][a-z]*[+-]*[0-9]*\]'
+    logger.info("Updating RDKit reactions")
     assert hasattr(RDKitReaction, "__table__")  # Type hint.
     table = RDKitReaction.__table__
     start = time.time()
+    session.execute(
+        insert(table)
+        .from_select(
+            ["reaction_smiles"],
+            select(Mappers.Reaction.reaction_smiles)
+            .join(Mappers.Dataset)
+            .where(Mappers.Dataset.dataset_id == dataset_id, Mappers.Reaction.reaction_smiles.is_not(None))
+            .distinct(),
+        )
+        .on_conflict_do_nothing(index_elements=["reaction_smiles"])
+    )
     session.execute(
         update(table)
         .where(table.c.reaction.is_(None))
         .values(reaction=func.reaction_from_smiles(cast(table.c.reaction_smiles, CString)))
     )
-    logger.info(f"Adding reaction took {time.time() - start}s")
-    logger.info("Populating RDKit mol columns")
+    logger.info(f"Updating reactions took {time.time() - start:g}s")
+    logger.info("Updating RDKit mols")
     assert hasattr(RDKitMol, "__table__")  # Type hint.
     table = RDKitMol.__table__
     start = time.time()
+    session.execute(
```

Comment:
What is the purpose of the 2nd `insert`?

Reply:
one is for `Compound` and the other is for `ProductCompound` SMILES